### PR TITLE
Deploy Octavia and Barbican to Production cluster

### PR DESCRIPTION
**Pull request by paulbrowne**
Deploy Octavia and Barbican to Production cluster, in service of deploying auto-scaling Magnum K8S clusters

### CODE REVIEW

File: etc/kayobe/kolla/config/octavia.conf [hunk 1]
```diff
@@ -0,0 +1,21 @@
+[certificates]
+cert_generator = local_cert_generator
+ca_certificate = /etc/octavia/certs/server_ca.cert.pem
+ca_private_key = /etc/octavia/certs/server_ca.key.pem
+ca_private_key_passphrase = <server CA key passphrase>
+
+[health_manager]
+controller_ip_port_list = 10.208.1.0:5555, 10.208.1.1:5555, 10.208.1.2:5555
```
Comment (JohnGarbutt) [inline etc/kayobe/kolla/config/octavia.conf:2]: Could we not template this... I think we have something similar for memcache here:
https://github.com/openstack/kolla-ansible/blob/e7a85726d23cdcf531fe0a82878d94314053cdfc/ansible/roles/octavia/templates/octavia.conf.j2#L36

**Review by JohnGarbutt — CHANGES_REQUESTED** [review]

File: etc/kayobe/kolla/config/octavia.conf (hunk 1 above)
Comment (paulbrowne) [reply etc/kayobe/kolla/config/octavia.conf:2]: In fact I see the very line concerned there too;
https://github.com/openstack/kolla-ansible/blob/e7a85726d23cdcf531fe0a82878d94314053cdfc/ansible/roles/octavia/templates/octavia.conf.j2#L56

I think that the issue here is that we're using a new dedicated network 10.208/16 for the Octavia control and sync network. which will be unknown to kolla-ansible at the moment.

If the Octavia HMs bind instead to the internal API network and amphorae could route to them instead, then that could be a potential way around modifying kolla-ansible.

Comment (JohnGarbutt) [reply etc/kayobe/kolla/config/octavia.conf:2]: Ah...

I suspect we need to add in octavia_interface into here somewhere:
https://github.com/openstack/kayobe/blob/aea7117764ee3ced0ccb7477c5c962569fa1c926/ansible/roles/kolla-ansible/defaults/main.yml#L104

But lets maybe just add a TODO to revisit that choice and fix it later.

**Review by JohnGarbutt — APPROVED** [review]

File: etc/kayobe/kolla/config/octavia.conf [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+client_ca = /etc/octavia/certs/client_ca.cert.pem
+amp_boot_network_list = ef9820af-ea57-4280-a43a-23c4ebd77bc6
+amp_image_tag = amphora
+amp_image_owner_id = 481837de9b39406d93c7782d0d29e48e
+amp_secgroup_list = a11521a8-ea17-4e2e-b8f8-45f152e2979c
+amp_flavor_id = 298272df-846a-4631-954a-e32823402286
+amp_ssh_key_name = octavia
+
+[haproxy_amphora]
+client_cert = /etc/octavia/certs/client.cert-and-key.pem
```
Comment (brtkwr) [inline etc/kayobe/kolla/config/octavia.conf]: where is the client.cert-and-key.pem?

Comment (paulbrowne) [reply etc/kayobe/kolla/config/octavia.conf]: Ah, that's left over from when I was looking at the dual CA model, need to take that out of the over-rides. The single CA key is Ansible-Vaulted.

**Pull request merged**